### DISCUSSION TITLE
Wildcards in hostlists?

### DISCUSSION THREAD

**Discussion by ENLinden** [evilsocket/opensnitch#461] [Q&A] [answered]
I saw that blocking traffic from a domain (i.e. `amazon.com`) can be prepended with `.*\.` (i.e. `.*\.amazon.com`) to also block all subdomains (i.e. `smile.amazon.com`, etc.). I'm aware that there's no way to do such wildcard blocking using the /etc/hosts file, and that it _can_ be done using dnsmasq, but I'd prefer to solve this problem using only opensnitch to keep things simple. 

It doesn't appear as though prepending `.*\.` to a domain has the same functionality when opensnitch is reading host rules from a file using the "list of domains" option, nor does it seem to be the case that it assumes all subdomains without the need for a wildcard character/string.

Is there an easy way to have opensnitch block all subdomains when reading a list of host rules from a file? If not, could this functionality be added? This is definitely mission-critical, I can't think of any feasible way to block as many URLs as I need to without this functionality.

**Comment by NRGLine4Sec** [comment]
I agree. This feature could be very convenient.

**Comment by gustavo-iniguez-goya** [comment, marked as answer]
Hi @ERLindeman ,

> Is there an easy way to have opensnitch block all subdomains when reading a list of host rules from a file? 

Unfortunately it's not possible to use wildcards in the hostlists at the moment. Actually I did think about it, but I was worried about the performance, so I prefered to add something that just worked for this version (1.4.0). Now that it seems to be working fine, I can play a little bit and test a PoC.

I'd also like to add peridoc updates of the lists. But for next version, 1.5.0 or so. 1.4.0 is about to be freezed and released.

**Reply by ENLinden** [reply]
If you could enable wildcard availability as an option in 1.5.0, that would be amazing. I'm looking to lock down my firewall settings soon, so the sooner that's available, the better!

**Reply by gustavo-iniguez-goya** [reply]
I've got this feature working. There's no much performance penalty with a dozens of hosts. It's cool because if you use a lists of domains like this one https://v.firebog.net/hosts/Admiral.txt the domain is a valid regexp as is, so for example for `test.com` it'll match www.test.com, www.test.com.localdomain, etc... 

I'll test it a little bit more and I'll push it.

**Reply by gustavo-iniguez-goya** [reply]
rereading the request, I don't know if you expect the wildcards to be used with lists of domains in "hosts" format, but that's not what I added. To be clear this is what is done right now:

* support for lists in hosts formats (0.0.0.0 example.com) (no wildcards allowed)
* support for list of IPs and network ranges.
* support for lists of domains (only 1 domain per line, without 127.0.0.1 or 0.0.0.0) with or without regular expressions: 
   *  example.com -> will match any domain that contains "example.com" in the domain name (like example.com, www.example.com, example.com.au, etc)
   * (example\.com)$ -> will match only *example.com
   * ^(example\.com)$ -> will match only example.com

**Reply by gustavo-iniguez-goya** [reply]
packages ready to test this feature: https://github.com/evilsocket/opensnitch/releases/tag/v1.5.0-rc.1
